File: Website/AdminSite/service/authenticate/refreshService.js
function refreshToken() {
  fetch("http://103.200.20.167:8083/cinema/refresh", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      token: sessionStorage.getItem("token"),
    }),
  })
    .then((response) => response.json())
    .then((data) => {
      if (data.code !== 1000) {
        console.log("Token refresh failed");
      } else {
        sessionStorage.setItem("token", data.result.token);
      }
    })
    .catch((error) => {
      console.error("Error refreshing token:", error);
    });
}
setInterval(() => {
  refreshToken();
}, 3600000);

File: App/movie_booking_app/lib/services/Users/refresh/refreshService.dart
import 'dart:convert';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:http/http.dart' as http;
import 'package:movie_booking_app/provider/sharedPreferences/prefs.dart';

class RefreshToken {
  static Future<dynamic> refreshToken() async {
    Preferences pref = Preferences();
    String? expriredToken = await pref.getTokenUsers();
    try {
      await dotenv.load();
      final getURL = dotenv.env['REFRESH_TOKEN']!;
      final url = getURL;
      final refreshBody = json.encode({'token': expriredToken});
      final response = await http.post(
        Uri.parse(url),
        headers: {
          'Content-type': 'application/json',
        },
        body: refreshBody,
      );

      final responseData = json.decode(response.body);
      if (responseData['code'] != 1000) {
        return;
      }
      dynamic newToken = responseData['result']['token'];
      pref.saveTokenKey(newToken);
      print('new token:$newToken');
    } catch (e) {
      return 'Error: $e';
    }
  }
}
